connect-examples/server-library-example/server/src/main/scala/org/example/CustomTable.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.example
+
+import java.util.UUID
+import com.github.mrpowers.spark.daria.sql.DariaWriters
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.apache.spark.sql.types.StructType
+
+/**
+ * Represents a custom table with an identifier and a DataFrame.
+ *
+ * @param identifier The unique identifier for the table.
+ * @param df The DataFrame associated with the table.
+ */
+class CustomTable private (identifier: CustomTable.Identifier, df: Dataset[Row]) {
+
+  /**
+   * Returns the DataFrame associated with the table.
+   *
+   * @return The DataFrame.
+   */
+  def toDF(): Dataset[Row] = df
+
+  /**
+   * Writes the DataFrame to disk as a CSV file.
+   */
+  def flush(): Unit = {
+    // Write dataset to disk as a CSV file
+    DariaWriters.writeSingleFile(
```

Comment:
How is this better than using dt.write?

Reply:
The regular write operation creates a folder and files in the format `mydata.csv/part-00000`. For simplicity, I figured writing a single '.csv' would work best (but comes at the cost of requiring the spark-daria jar here)